### DISCUSSION TITLE
Question on Point Sources

### DISCUSSION THREAD

**Discussion by rwalkerlewis** [idaholab/moose#18044] [Q&A Modules: General] [answered]
Hello,

I am having a bit of trouble understanding exactly how the mass flux (mass over time) either inputted or generated (such as in the Peaceman Borehole case) relates to the source term, q (defined as density over time). How is the unit discrepancy resolved?

Thank you,

Robert

**Comment by WilkAndy** [comment, marked as answer]
Mathematically it is because the delta function has units m^{-3}.  This is mentioned after Eqn(1) of https://mooseframework.inl.gov/modules/porous_flow/sinks.html .   I'm not sure whether this answers your question, or whether you've read the documentation and that's what caused your confusion!  Please tell me if you're still confused and i'll attempt to explain further.

**Reply by hugary1995** [reply]
~I think the dirac delta function is dimensionless actually. 
s in your Eqn(1) actually has units of kg.s^{-1}. It gets "sifted" out from the volume integral due to the dirac delta function.
For example, (\int f(x) \delta(x) dV) has the same units as f.~

~I guess it's okay to say dirac delta has units of m^{-3} if you consider some sort of regularized measure but that'll depend on your spatial dimensionality then...~

Nope, I lied -- see below. I wanted to say \delta * dV is dimensionless.

**Reply by WilkAndy** [reply]
Mathematically the delta function has units 1/V, and yes, it depends on the dimensionality (so m^{-2} in 2D problems, etc).  See, eg, https://physics.stackexchange.com/questions/33760/what-are-the-units-or-dimensions-of-the-dirac-delta-function .  So the units in Eqn(1) and the following discussion are correct.

But, oh dear, I think this explanation involving delta functions might not be enlightening to @rwalkerlewis .   Would it be helpful to explain a bit further, @rwalkerlewis  ?

**Reply by hugary1995** [reply]
Okay, you are right. Thanks,
